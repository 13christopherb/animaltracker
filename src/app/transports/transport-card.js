import React from "react";
import {TransportListItem} from "./transport-list-item";


export const TransportCard = ({transports}) => {
    return (
            <ul className="list-group">
                {transports.map(transport => <TransportListItem key={transport.id} transport={transport} />)}
            </ul>
    )
};