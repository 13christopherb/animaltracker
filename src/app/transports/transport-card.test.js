import React from 'react';
import {shallow} from 'enzyme';
import {TransportCard} from "./transport-card";
import moment from 'moment';
import toJson from 'enzyme-to-json';

const transports = [
    {
        departs: 'MBO',
        arrives: 'NRO',
        meetTime: moment('2017-09-15 09:30:00')
    },
    {
        departs: 'SLO',
        arrives: 'MBO',
        meetTime: moment('2017-09-15 011:30:00')
    }
];

describe('<TransportCard />', () => {
    describe('render()', () => {
        test('should render the component', () => {
            const wrapper = shallow(<TransportCard transports={transports}/>);

            expect(toJson(wrapper)).toMatchSnapshot()
        })
    });
});