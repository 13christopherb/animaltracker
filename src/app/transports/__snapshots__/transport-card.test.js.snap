// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransportCard /> render() should render the component 1`] = `
<ul
  className="list-group"
>
  <TransportListItem
    transport={
      Object {
        "arrives": "NRO",
        "departs": "MBO",
        "meetTime": "2017-09-15T16:30:00.000Z",
      }
    }
  />
  <TransportListItem
    transport={
      Object {
        "arrives": "MBO",
        "departs": "SLO",
        "meetTime": "2017-09-15T18:30:00.000Z",
      }
    }
  />
</ul>
`;
